data-pipeline/content-cache-updater/src/main/java/org/ekstep/ep/samza/service/ContentCacheUpdaterService.java

```diff
@@ -0,0 +1,153 @@
+import redis.clients.jedis.exceptions.JedisException;
+
+import java.lang.reflect.Type;
+import java.util.*;
+
+public class ContentCacheUpdaterService {
+
+    private static Logger LOGGER = new Logger(ContentCacheUpdaterService.class);
+    private RedisConnect redisConnect;
+    private Jedis dialCodeStoreConnection;
+    private Jedis contentStoreConnection;
+    private JobMetrics metrics;
+    private int dialCodeStoreDb;
+    private int contentStoreDb;
+    Jedis redisConnection;
+    int storeId;
+    private Gson gson = new Gson();
+    Type mapType = new TypeToken<Map<String, Object>>() {
+    }.getType();
+    private List<String> contentModelListTypeFields;
+
+    public ContentCacheUpdaterService(Config config, RedisConnect redisConnect, JobMetrics metrics) {
+        this.redisConnect = redisConnect;
+        this.metrics = metrics;
+        this.contentStoreDb = config.getInt("redis.database.contentStore.id", 5);
+        this.dialCodeStoreDb = config.getInt("redis.database.dialCodeStore.id", 6);
+        this.contentStoreConnection = redisConnect.getConnection(contentStoreDb);
+        this.dialCodeStoreConnection = redisConnect.getConnection(dialCodeStoreDb);
+        this.contentModelListTypeFields = config.getList("contentModel.fields.listType", new ArrayList<>());
+    }
+
+    public void process(ContentCacheUpdaterSource source, ContentCacheUpdaterSink sink) {
+        Map<String, Object> message = source.getMap();
+        String nodeUniqueId = (String) message.get("nodeUniqueId");
+        String objectType = (String) message.get("objectType");
+        Map<String, Object> parsedData = null;
+        if (nodeUniqueId == null || objectType == null) return;
+        LOGGER.info("", "processing event for nodeUniqueId: " + nodeUniqueId);
+        try {
+            if (!nodeUniqueId.isEmpty() && objectType.equalsIgnoreCase("DialCode")) {
+                parsedData = updateDialCodeCache(message, sink);
+                redisConnection=dialCodeStoreConnection;
+                storeId=dialCodeStoreDb;
+            } else if (!nodeUniqueId.isEmpty()) {
+                parsedData = updateContentCache(message, sink);
+                redisConnection=contentStoreConnection;
+                storeId=contentStoreDb;
+            }
+            if(null != parsedData) {
+                redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+                sink.success();
+            }
+        }catch (JedisException ex) {
+            metrics.incCacheErrorCounter();
+            LOGGER.error("", "Exception when adding to dialcode redis cache", ex);
+            redisConnect.resetConnection();
+            try (Jedis redisConn = redisConnect.getConnection(storeId)) {
+                this.redisConnection = redisConn;
+                if (null != parsedData)
+                    redisConnect.addJsonToCache(nodeUniqueId, gson.toJson(parsedData), redisConnection);
+            }
+        }
```

Comment:
Move this code a base class. Create a BaseCacheUpdater under the common library and add the methods such as addToCache(), readFromCache() etc with retry logic built in